National Library of Norway.js

```diff
@@ -35,42 +35,99 @@
 
 	***** END LICENSE BLOCK *****
 */
-
 function detectWeb(doc, url) {
-	if (url.indexOf("/nbsok/search?") != -1) {
-		return "multiple";
-	} else if (url.indexOf("/nbsok/nb/") != -1) {
-		var mediaTypes = {
-			'Bøker': 'book',
-			'Tidsskrift': 'book', // complete issues or volumes, not articles
-			'Aviser': 'newspaperArticle',
-			'Film': 'videoRecording',
-			'Fjernsyn': 'videoRecording',
-			'Radio': 'radioBroadcast',
-			'Kart': 'map',
-		};
-		var nodes = ZU.xpath(doc, '//input[@id="mediaType"]');
-		if (nodes.length && mediaTypes[nodes[0].value]) {
-			return mediaTypes[nodes[0].value];
+	// The page can change from a search page to a single item page
+	// without loading the whole content as a new website, so we
+	// need to monitor for DOM changes.
+	Z.monitorDOMChanges(ZU.xpath(doc, '//div[contains(@class, "layout-content")]')[0], {childList: true});
+
+	// New UI - search page
+	if (url.indexOf('nb.no/search?') != -1){
+		return 'multiple';
+	}
+
+	// Old UI – item (these URLs are not yet redirected to the new UI)
+	if (url.indexOf('nb.no/nbsok/nb/') != -1){
+		var nodes = ZU.xpath(doc, '//meta[@name="dc:type"]');
+		if (nodes.length) {
+			var dcType = nodes[0].getAttribute('content');
+			return mapMediaType(dcType);
 		}
 	}
+
+	// New UI - item
+	if (url.indexOf('nb.no/items/') != -1){
+		// There is a dc:type meta tag, but it contains values like
+		// "nonfiction" or "Text", so not helpful like before when it contained
+		// material type.
+		return 'book';
+	}
 }
 
 function doWeb(doc, url) {
-	if (detectWeb(doc, url) == "multiple") {
-		var items = ZU.getItemArray(doc, doc, '/nbsok/nb/');
+	if (detectWeb(doc, url) == 'multiple') {
+		var linkItems = ZU.xpath(doc, '//mat-card/a[contains(@href, "items")]');
+		var items = {};
+		linkItems.forEach(function(linkItem) {
+			var link = linkItem.getAttribute('href');
+			var title = '?';
+			var elems = ZU.xpath(linkItem, './/label[@class="title"]');
+			var dds = ZU.xpath(linkItem, './/dd').slice(0, 3);  // Limit to first three
+
+			// The search result list has two different layouts
+			if (elems.length) {
+				// Grid view
+				title = elems[0].textContent;
+				ZU.xpath(linkItem, './/label[@class="subtitle"]').forEach(function(elem) {
+					title += ' / ' + elem.textContent;
+				});
+			} else if (dds.length) {
+				// List view
+				title = dds[0].textContent;
+				dds.forEach(function(dd) {
+					title += ' / ' + dd.textContent;
+				});
+			}
+			items[link] = title;
```

Comment:
test for presence of title & link here to be safe.

Reply:
Done